pkg/credential/token.go

```diff
@@ -30,12 +35,49 @@ type GcpTokenFetcher struct {
 	Aud string
 }
 
+// AwsTokenFetcher implements the token fetcher in AWS.
+// It gets the EC2 instance identity document as described
+// in the doc: http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+type AwsTokenFetcher struct {
+}
+
+// IsOnAWS returns if node agent is running on AWS EC2 instance
+func (fetcher *AwsTokenFetcher) IsOnAWS() bool {
```

Comment:
Make IsOnAWS private?

Reply:
this is called by `awsPlatformImpl.IsProperPlatform()`. Will move it to better place.